Library/Homebrew/cmd/update.rb

```diff
@@ -277,8 +277,7 @@ def pull!(options = {})
     reset_on_interrupt { safe_system "git", *args }
 
     @current_revision = read_current_revision
-
-    if @initial_branch != @upstream_branch && !@initial_branch.empty?
+    if ARGV.homebrew_developer? && @initial_branch != @upstream_branch && !@initial_branch.empty?
```

Comment:
Perhaps we should just remove this logic altogether for everyone. I don't think changing back to a branch is too much work.